src/mpi/datatype/typerep/src/typerep_yaksa_pack.c

```diff
@@ -163,7 +163,11 @@ static int typerep_do_pack(const void *inbuf, MPI_Aint incount, MPI_Datatype dat
         MPI_Aint real_bytes = MPL_MIN(total_size - inoffset, max_pack_bytes);
         /* Make sure we never pack partial element */
         real_bytes -= real_bytes % element_size;
-        MPIR_Memcpy(outbuf, MPIR_get_contig_ptr(inbuf_ptr, inoffset), real_bytes);
+        if (flags & MPIR_TYPEREP_FLAG_STREAM) {
+            MPL_Memcpy_stream(outbuf, MPIR_get_contig_ptr(inbuf_ptr, inoffset), real_bytes);
+        } else {
+            MPIR_Memcpy(outbuf, MPIR_get_contig_ptr(inbuf_ptr, inoffset), real_bytes);
+        }
```

Comment:
There is an asymmetry here -- `MPL_` and `MPIR_` . Probably should fold `MPL_Memcpy_stream` inside `MPIR_Memcpy`.

Reply:
Yeah, I went back and forth on this a bunch of times. I ended up going with MPL because it was more portable that way and the implementation of `MPIR_Memcpy` is a very, very small shim around the existing `memcpy`:

https://github.com/pmodels/mpich/blob/3e44a99995599c15e9c269f0f984c0b89833f7f7/src/include/mpir_mem.h#L81-L85

It seemed like adding additional logic here was probably not what we would want long term. Happy to hear other thoughts here.

Reply:
I'm not sure if I prefer MPL or MPIR, but I do know I don't like the split :smile:. MPL is probably a better place in theory, but I'm not sure churning all existing MPIR_Memcpy->MPL_memcpy is worth it...

Reply:
I agree that the current status here is ugly. MPL certainly seems like the right place to me, but I think that's a separate issue from this PR. At least right now, MPL has `aligned_alloc`, `calloc`, `malloc`, `realloc`, and 'free' and all of these are implemented as functions. On the other hand MPIR has a bunch of macro wrappers that call over to MPL when they're done. I don't think this new stuff should be implemented as a macro.

Reply:
I added a couple of extra fixup patches at the end of this branch that move the streaming copy code into MPIR instead of MPL. Take a look and see what you think.